### DISCUSSION TITLE
Does autoDispose individually dispose each family member?

### DISCUSSION THREAD

**Discussion by mark8044** [rrousselGit/riverpod#1429] [Q&A] [answered]
I have a `family` of `ChangeNotifierProvider` with `autoDispose` enabled

```
final newsProvider = ChangeNotifierProvider.autoDispose.family<NewsData, String>((ref, id) {
  return NewsData();
});
```

Each `family` member, holds data for a given news article.

The entire `newsProvider` is constantly being listened to, but the individual family members get loaded in at different times as a person decides to look at a news article.  

When the individual news article is closed, the individual family member is not being listened to anymore

```
final data = ref.watch(newsProvider(familymemberID)).map;
```

Will autoDispose then clean out the data for that individual family member and leave the other family members untouched if they are still being listened to?   or does the entire `newsProvider` have to get unlistened to for the entire family to get autoDispose'd ?

**Comment by rrousselGit** [comment, marked as answer]
Yes, autoDispose applies to individual parameters

If one specific ID is no-longer used, then it'll get disposed,
independently from other IDs

**Reply by mark8044** [reply]
Too perfect!  Thank you
